modin/core/dataframe/pandas/partitioning/partition.py

```diff
@@ -38,6 +38,24 @@ class PandasDataframePartition(ABC):  # pragma: no cover
     _width_cache = None
     _identity_cache = None
     _data = None
+    execution_wrapper = None
+
+    # these variables are intentionally initialized at runtime
+    # so as not to initialize the engine during import
+    _iloc_func = None
+
+    def __init__(self):
+        if type(self)._iloc_func is None:
+            # Places `_iloc` function into the storage to speed up
+            # remote function calls and caches the result.
+            # It also postpones engine initialization, which happens
+            # implicitly when `execution_wrapper.put` is called.
+            if self.execution_wrapper is not None:
+                type(self)._iloc_func = staticmethod(
+                    self.execution_wrapper.put(self._iloc)
+                )
+            else:
+                type(self)._iloc_func = staticmethod(self._iloc)
```

Comment:
```suggestion
                type(self)._iloc_func = self._iloc
```
self._iloc is already a static method. Why do we need to wrap it over again?

Reply:
Consider the following example:
```python
>>> class A:
...   @staticmethod
...   def test_func():
...     print("call")
...
>>> A.test_func
<function A.test_func at 0x0000015615BAE280>
>>> A.test_func()
call
>>> A().test_func()
call
>>> A.test_func2 = A().test_func  # emulate behavior as in __init__
>>> A.test_func2()
call
>>> A().test_func2()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: test_func() takes 0 positional arguments but 1 was given
>>> A.test_func2 = staticmethod(A().test_func)
>>> A.test_func2()
call
>>> A().test_func2()
call
```

Reply:
I see, thanks!